specification/keyvault/data-plane/TypeSpec/models.tsp

```diff
@@ -0,0 +1,389 @@
+   * retention interval (90 days), immediate and permanent deletion (i.e. purge) is
+   * not permitted, and in which the subscription itself  cannot be permanently
+   * canceled. System wil permanently delete it after 90 days, if not recovered
+   */
+  Recoverable_ProtectedSubscription: "Recoverable+ProtectedSubscription",
+
+  /**
+   * Denotes a vault state in which deletion is recoverable, and which also permits
+   * immediate and permanent deletion (i.e. purge when 7<= SoftDeleteRetentionInDays
+   * < 90). This level guarantees the recoverability of the deleted entity during
+   * the retention interval, unless a Purge operation is requested, or the
+   * subscription is cancelled.
+   */
+  CustomizedRecoverable_Purgeable: "CustomizedRecoverable+Purgeable",
+
+  /**
+   * Denotes a vault state in which deletion is recoverable without the possibility
+   * for immediate and permanent deletion (i.e. purge when 7<=
+   * SoftDeleteRetentionInDays < 90).This level guarantees the recoverability of the
+   * deleted entity during the retention interval and while the subscription is
+   * still available.
+   */
+  CustomizedRecoverable: "CustomizedRecoverable",
+
+  /**
+   * Denotes a vault and subscription state in which deletion is recoverable,
+   * immediate and permanent deletion (i.e. purge) is not permitted, and in which
+   * the subscription itself cannot be permanently canceled when 7<=
+   * SoftDeleteRetentionInDays < 90. This level guarantees the recoverability of the
+   * deleted entity during the retention interval, and also reflects the fact that
+   * the subscription itself cannot be cancelled.
+   */
+  CustomizedRecoverable_ProtectedSubscription: "CustomizedRecoverable+ProtectedSubscription",
+}
+
+/**
+ * The secret set parameters.
+ */
+model SecretSetParameters {
+  /**
+   * The value of the secret.
+   */
+  value: string;
+
+  /**
+   * Application specific metadata in the form of key-value pairs.
+   */
+  tags?: Record<string>;
+
+  /**
+   * Type of the secret value such as a password.
+   */
+  contentType?: string;
+
+  /**
+   * The secret management attributes.
+   */
+  @encodedName("application/json", "attributes")
+  secretAttributes?: SecretAttributes;
+}
+
+/**
+ * The secret management attributes.
+ */
+model SecretAttributes extends Attributes {
+  /**
+   * softDelete data retention days. Value should be >=7 and <=90 when softDelete
+   * enabled, otherwise 0.
+   */
+  @visibility("read")
+  recoverableDays?: int32;
+
+  /**
+   * Reflects the deletion recovery level currently in effect for secrets in the
+   * current vault. If it contains 'Purgeable', the secret can be permanently
+   * deleted by a privileged user; otherwise, only the system can purge the secret,
+   * at the end of the retention interval.
+   */
+  @visibility("read")
+  recoveryLevel?: DeletionRecoveryLevel;
+}
+
+/**
+ * The object attributes managed by the KeyVault service.
+ */
+model Attributes {
```

Comment:
Where is this used other than as a base class?

Reply:
This is just used as a base class that comes from the original swagger.